worldwind/src/commonMain/kotlin/earth/worldwind/draw/DrawableShape.kt

```diff
@@ -64,48 +63,23 @@ open class DrawableShape protected constructor(): Drawable {
         // Make multi-texture unit 0 active.
         dc.activeTextureUnit(GL_TEXTURE0)
 
-        // Use the shape's vertex point attribute and vertex texture coordinate attribute.
-        dc.gl.enableVertexAttribArray(1 /*vertexTexCoord*/)
-        dc.gl.enableVertexAttribArray(2 /*vertexTexCoord*/)
-        dc.gl.enableVertexAttribArray(3 /*vertexTexCoord*/)
+        program.enableLinesMode(drawState.isLine)
+        program.enableVertexColorAndWidth(drawState.isStatic)
+        program.loadScreen(dc.viewport.width.toFloat(), dc.viewport.height.toFloat())
 
-        if (drawState.isLine) {
-            program.enableOneVertexMode(false)
-            program.loadScreen(dc.viewport.width.toFloat(), dc.viewport.height.toFloat())
-            dc.gl.vertexAttribPointer(0 /*pointA*/, 4, GL_FLOAT, false, 20, 0)
-            dc.gl.vertexAttribPointer(1 /*pointB*/, 4, GL_FLOAT, false, 20, 40)
-            dc.gl.vertexAttribPointer(2 /*pointC*/, 4, GL_FLOAT, false, 20, 80)
-            dc.gl.vertexAttribPointer(3 /*texCoord*/, 1, GL_FLOAT, false, 20, 56)
-        } else {
-            program.enableOneVertexMode(true)
-            dc.gl.vertexAttribPointer(0 /*vertexPoint*/, 3, GL_FLOAT, false, drawState.vertexStride, 0)
-            dc.gl.vertexAttribPointer(1 /*vertexPoint*/, 3, GL_FLOAT, false, drawState.vertexStride, 0)
-            dc.gl.vertexAttribPointer(2 /*vertexPoint*/, 3, GL_FLOAT, false, drawState.vertexStride, 0)
-        }
         // Draw the specified primitives.
         for (idx in 0 until drawState.primCount) {
             val prim = drawState.prims[idx]
-            program.loadColor(prim.color)
             program.loadOpacity(prim.opacity)
+            program.loadColor(prim.color)
+            program.loadLineWidth(prim.lineWidth)
             if (prim.texture?.bindTexture(dc) == true) {
                 program.loadTexCoordMatrix(prim.texCoordMatrix)
                 program.enableTexture(true)
             } else {
                 program.enableTexture(false)
             }
-            if (drawState.isLine) {
-                program.loadLineWidth(prim.lineWidth)
-            } else {
-                dc.gl.vertexAttribPointer(
-                    3 /*vertexTexCoord*/,
-                    prim.texCoordAttrib.size,
-                    GL_FLOAT,
-                    false,
-                    drawState.vertexStride,
-                    prim.texCoordAttrib.offset
-                )
-                dc.gl.lineWidth(prim.lineWidth)
-            }
+            dc.gl.lineWidth(prim.lineWidth)
```

Comment:
Makes no sense to call this unconditionally

Reply:
done